File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest54523.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest54523 {

    private static String normalize(String v) { return v.strip(); }

    @PostMapping(path="/BenchmarkTest54523", consumes="application/xml")
    public void BenchmarkTest54523(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        String data = normalize(xmlValue);
        response.setHeader("X-Frame-Options", "DENY");
        response.setHeader("Content-Security-Policy", "frame-ancestors 'none'");
        response.getWriter().print(String.valueOf(data));
    }
}

File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_54523.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request
from app_runtime import db


async def BenchmarkTest54523(request: Request):
    multipart_value = (await request.form()).get('multipart_field', '')
    data = (lambda v: v.strip())(multipart_value)
    db.users.find({'$where': "this.username == '" + str(data) + "'"})
    return {"updated": True}
